p2p/src/interface/event_channel.rs

```diff
@@ -0,0 +1,299 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use futures::{channel::mpsc, Sink, SinkExt, Stream};
+use pin_project::pin_project;
+use std::{
+    collections::VecDeque,
+    pin::Pin,
+    task::{Context, Poll, Waker},
+};
+
+/// Configure how the network should behave in the case that the capacity of the [`EventChannel`] is reached, i.g. if
+/// the channel is full. This is relevant in cases when the frequency of messages is larger than the frequency
+/// in which the [`mpsc::Receiver`][futures::channel::mpsc::Receiver] side reads from the stream.
+pub enum ChannelSinkConfig {
+    /// Block until the channel has enough capacity for the new request.
+    ///
+    /// **Note**: This pauses all network interaction and enforces back-pressure, which may be desirable if the machine
+    /// is at its limit. But it also hinders all active actions on `StrongholdP2p`, hence asynchronous methods like
+    /// [`StrongholdP2p::send_request`][super::StrongholdP2p::send_request] will be blocked as well.
+    Block,
+    /// New events will be dropped if the channel is full.
+    DropLatest,
+    /// In case that the channel is full, store new events in a ring-buffer. If the configured capacity is reached,
+    /// older events will be dropped in favor of newer ones. Send the latest events sequentially in FIFO order when
+    /// the channel has free capacity.
+    BufferLatest,
+}
+
+/// Wrapper of a [`mpsc::channel`][futures::channel::mpsc::channel] for sending events.
+/// The [`EventChannel`] cen be configure its behaviour e capacity of the channel is reached.
```

Comment:
sorry, I do not understand this comment. Is this meant:
"The [`EventChannel`] can configure its behavior until the capacity of the channel is reached." ?

Reply:
I don't understand it myself either. Must have been my cat walking over the keyboard 😄.